final-project/manRoomStates.tex

```diff
@@ -0,0 +1,21 @@
+\chapter{Maintaining Room State \\
+  \small{\textit{-- Evan Ciok, Sophia DiCuffa, Carson McManus}}
+  \index{Chapter!room-state}
+  \label{Chapter::RoomState}}
+
+\section{Lost Balancer-Monolith Connections}
+
+The connection between the load balancer and the individual monoliths is vital to the operation of OTT, so when one of these connections is lost, it needs to be handled appropriately. 
+The most important consideration is what to do with all clients connected to rooms on that given monolith. Without the balancer connecting to it, it is no longer viable and all clients 
+should be unloaded from the rooms. From there, they can reconnect to a room through the load balancer to a monolith with a valid connection. 
+
+\section{Duplicate Rooms Across Monoliths}
+
+Two monolith nodes should never have the same room loaded, as the load balancer should be directing all connections for a given room to the designated monolith using its preserved states. 
+In the case that this does happen, the system is then in a bad state and it must be resolved. The proposed solution to this issue is to have the load balancer unload rooms that were not the 
+first instance of that particular room. This means that the duplicate instances which put the system into a bad state would be unloaded and the clients could then reconnect to the room with 
+the proper path. In order to accomplish this, two approaches have been hypothesized thus far. First, every room could have a timestamp associated with it, designating the time it was loaded. 
+Then, in the case of duplication, the timestamps can be compared and the more recently loaded instances can be handled appropriately. The other potential approach is to store an atomic integer 
+in redis that is automatically incremented upon each room load, which can be used to appropriately determine the original instance of a room if needed. No matter which approach is taken, 
+the key issue is flagging the original instance and unloading the rest to return the system to a healthy state. An important consideration in this implementation is assessing what happens to 
```

Comment:
> An important consideration

Again, you are specifying the project. You need to consider this, and write down what should and should not happen, not leave it to the reader to decide.